### PR TITLE
In-memory sqlite locking & improved image handling 

### PR DESCRIPTION
**Pull request by guusw**
- Image ref-counting
- Loading RGB (no alpha) images directly into gfx::Texture
- Add in-memory locking mechanism for sqlite (on emscripten)

### CODE REVIEW

File: include/shards/shards.h [hunk 1]
```diff
@@ -233,13 +233,19 @@ struct SHColor {
 #define SHIMAGE_FLAGS_16BITS_INT (1 << 2)
 #define SHIMAGE_FLAGS_32BITS_FLOAT (1 << 3)
 
+struct SHImage;
+typedef void(__cdecl *SHImageFreeProc)(struct SHImage *image);
+
 struct SHImage {
```
Comment (sinkingsugar) [inline include/shards/shards.h:239]: Can you quickly check the swift shards to see if broke?

Comment (sinkingsugar) [reply include/shards/shards.h:239]: I checked, there is nothing for image

File: shards/core/async.hpp [hunk 2]
```diff
@@ -22,12 +22,16 @@
 #endif
 
 // Can not create this many workers, create on demand instead
-#if SH_EMSCRIPTEN
+#if SH_EMSCRIPTEN || 1
```
Comment (sinkingsugar) [inline shards/core/async.hpp]: fix this

File: shards/lang/src/eval.rs [hunk 3]
```diff
@@ -59,7 +59,7 @@ pub(crate) const EVAL_STACK_SIZE: usize = 2 * 1024 * 1024;
 #[cfg(debug_assertions)]
 const MIN_STACK_SIZE: i64 = 2 * 1024 * 1024; // 2 MB for debug builds
 #[cfg(not(debug_assertions))]
-const MIN_STACK_SIZE: i64 = 64 * 1024; // 64 KB for release builds
+const MIN_STACK_SIZE: i64 = 128 * 1024; // 64 KB for release builds
```
Comment (sinkingsugar) [inline shards/lang/src/eval.rs]: revert

File: shards/modules/desktop/desktop.win.cpp [hunk 4]
```diff
@@ -477,19 +477,19 @@ struct Pixels : public PixelBase {
...
 
-  SHVar _output;
+  OwnedVar _output;
 
   Pixels() {
-    _output = SHVar(); // zero it
-    _output.valueType = SHType::Image;
-    _output.payload.imageValue.channels = 4;
-    _output.payload.imageValue.flags = 0;
+    // _output = SHVar(); // zero it
```
Comment (sinkingsugar) [inline shards/modules/desktop/desktop.win.cpp]: don't need this?

File: shards/modules/gfx/texture.cpp [hunk 5]
```diff
@@ -461,8 +453,8 @@ struct ReadTextureShard {
   RequiredGraphicsContext _requiredGraphicsContext;
 
   GpuTextureReadBufferPtr _readBuffer = makeGpuTextureReadBuffer();
-  SHVar _image;
-  std::vector<uint8_t> _imageBuffer;
+  OwnedVar _image;
+  // std::vector<uint8_t> _imageBuffer;
```
Comment (sinkingsugar) [inline shards/modules/gfx/texture.cpp]: remove?

File: shards/modules/sqlite/sqlite.cpp [hunk 6]
```diff
@@ -29,9 +31,9 @@ extern "C" {
 int sqlite3_crsqlite_init(sqlite3 *db, char **pzErrMsg, const sqlite3_api_routines *pApi);
 }
 
-static const char *vfs =
-#if SH_EMSCRIPTEN
-    "unix-flock";
+static const char *vfs = //
+#if SH_EMSCRIPTEN || 1
```
Comment (sinkingsugar) [inline shards/modules/sqlite/sqlite.cpp]: fix this, remove 1

Comment (sinkingsugar) [reply shards/modules/sqlite/sqlite.cpp]: remove if it's not the issue. prefer default implementations.

File: shards/rust/src/types.rs [hunk 7]
```diff
@@ -1700,16 +1701,24 @@ impl<'de> Deserialize<'de> for ClonedVar {
...
-            let height: u16 = seq.next_element()?.unwrap();
-            let channels: u8 = seq.next_element()?.unwrap();
-            let flags: u8 = seq.next_element()?.unwrap();
-            let value: &[u8] = seq.next_element()?.unwrap();
-            v.payload.__bindgen_anon_1.imageValue.width = width;
-            v.payload.__bindgen_anon_1.imageValue.height = height;
-            v.payload.__bindgen_anon_1.imageValue.channels = channels;
-            v.payload.__bindgen_anon_1.imageValue.flags = flags;
-            v.payload.__bindgen_anon_1.imageValue.data = value.as_ptr() as *mut u8;
+            // let width: u16 = seq.next_element()?.unwrap();
```
Comment (sinkingsugar) [inline shards/rust/src/types.rs]: might as well implement it?

File: shards/tests/rust.shs [hunk 8]
```diff
@@ -56,11 +56,6 @@
   b | Hash.Sha2-256 | AppendTo(a)
   a | ToBase58 | Log | Assert.Is("QmZ4tDuvesekSs4qM5ZBKpXiZGun7S2CYtEZRB3DYXkjGx" true)
 
-  "../../assets/Freesample.svg"
```
Comment (sinkingsugar) [inline shards/tests/rust.shs:59]: why remove?

**Review by sinkingsugar — APPROVED** [review]
Approved but many points to fix I marked.

File: shards/modules/sqlite/sqlite.cpp (hunk 6 above)
Comment (guusw) [reply shards/modules/sqlite/sqlite.cpp]: Let me try

File: shards/tests/rust.shs (hunk 8 above)
Comment (guusw) [reply shards/tests/rust.shs:59]: Moved to image test

**Pull request merged**